cpp/src/reductions/minmax.cu

```diff
@@ -98,35 +150,41 @@ struct minmax_functor {
   std::pair<std::unique_ptr<scalar>, std::unique_ptr<scalar>> operator()(
     const cudf::column_view &col, rmm::mr::device_memory_resource *mr, cudaStream_t stream)
   {
-    auto device_col = column_device_view::create(col, stream);
+    auto device_col  = column_device_view::create(col, stream);
+    using OutputType = minmax_pair<T>;
 
     // compute minimum and maximum values
-    minmax_pair<T> const result = [&]() -> minmax_pair<T> {
+    auto dev_result = [&]() -> rmm::device_scalar<OutputType> {
       auto op = minmax_iterfunctor<T>{*device_col};
       auto begin =
         thrust::make_transform_iterator(thrust::make_counting_iterator<size_type>(0), op);
-      auto end = begin + col.size();
 
       if (col.nullable()) {
-        return thrust::reduce(rmm::exec_policy(stream)->on(stream),
-                              begin,
-                              end,
-                              minmax_pair<T>{},
-                              minmax_binary_op<T, true>{});
+        auto binary_op = minmax_binary_op<T, true>{};
+        return reduce_device(begin, col.size(), binary_op, mr, stream);
       } else {
-        return thrust::reduce(rmm::exec_policy(stream)->on(stream),
-                              begin,
-                              end,
-                              minmax_pair<T>{},
-                              minmax_binary_op<T, false>{});
+        auto binary_op = minmax_binary_op<T, false>{};
+        return reduce_device(begin, col.size(), binary_op, mr, stream);
       }
     }();
 
-    std::unique_ptr<scalar> min =
-      make_fixed_width_scalar<T>(result.min_val, result.min_valid, stream, mr);
-    std::unique_ptr<scalar> max =
-      make_fixed_width_scalar<T>(result.max_val, result.max_valid, stream, mr);
-    return {std::move(min), std::move(max)};
+    using ScalarType = cudf::scalar_type_t<T>;
+    auto min         = new ScalarType(T{}, false, stream, mr);
+    auto max         = new ScalarType(T{}, false, stream, mr);
+
+    device_single_thread(
```

Comment:
What is `device_single_thread`?

Reply:
@karthikeyann I can look it up and guess, but are you able to answer this for Jake?

Reply:
`device_single_thread` launches single thread in GPU. (Useful for modifying single scalar)